### PR TITLE
feat: add support for layouting in rjsf forms 

### PR DESCRIPTION
**Pull request by lallenfrancisl**
To build intuition for this approach reading this issue will be helpful https://github.com/rjsf-team/react-jsonschema-form/issues/3503

Also read these to have a further idea on how all this works

1. https://rjsf-team.github.io/react-jsonschema-form/docs/advanced-customization/custom-templates#objectfieldtemplate
2. https://rjsf-team.github.io/react-jsonschema-form/docs/advanced-customization/custom-templates#fieldtemplate

So basically what this does is we will be is proxying the @rjsf/chakra-ui package by replacing components from them with ours.

An end point would be us building a similar package to @rjsf/chakra-ui with whatever support we need. I have referred mostly the source code from that package itself https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/chakra-ui/src/ObjectFieldTemplate/ObjectFieldTemplate.tsx

image

### CODE REVIEW

File: src/views/Connectors/Sources/SourcesForm/SourceConfigForm/SourceConfigForm.tsx [hunk 1]
```diff
@@ -57,4 +61,26 @@ const SourceConfigForm = (): JSX.Element | null => {
   );
 };
 
+const uiSchema = {
+  "ui:order": ["host", "port", "database", "credentials", "schema"],
+  "ui:layout": {
```
Comment (lallenfrancisl) [inline src/views/Connectors/Sources/SourcesForm/SourceConfigForm/SourceConfigForm.tsx:66]: This data is parsed in our custom ObjectFieldTemplate to specify the layout for these parts

File: src/views/Connectors/Sources/SourcesForm/SourceConfigForm/MWObjectFieldTemplate.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  if (layout?.display !== "grid" || !layout?.cols || !Array.isArray(layout?.colspans)) {
+    return <DefaultObjectFieldTemplate {...props} />
+  }
+
+  // Button templates are not overridden in the uiSchema
+  const {
+    ButtonTemplates: { AddButton },
+  } = registry.templates
+  const gridTemplate = `repeat(${layout.cols}, 1fr)`
```
Comment (lallenfrancisl) [inline src/views/Connectors/Sources/SourcesForm/SourceConfigForm/MWObjectFieldTemplate.tsx:58]: This is how the layout is set, since we have the data from the ui schema we can do this now

File: src/views/Connectors/Sources/SourcesForm/SourceConfigForm/SourceConfigForm.tsx [hunk 3]
```diff
@@ -45,8 +47,10 @@ const SourceConfigForm = (): JSX.Element | null => {
       <ContentContainer>
         <Box backgroundColor="gray.200" padding="20px" borderRadius="8px">
           <Form
+            uiSchema={(connectorSchema as any).title.toLowerCase() === "amazon redshift" ? uiSchema : undefined}
```
Comment (sumitd94) [inline src/views/Connectors/Sources/SourcesForm/SourceConfigForm/SourceConfigForm.tsx:50]: why is the type `any` here?

Comment (lallenfrancisl) [reply src/views/Connectors/Sources/SourcesForm/SourceConfigForm/SourceConfigForm.tsx]: I just added it to demonstrate how we are fixing this layout issue, so that code is not even needed so I just slapped `any` to fix the type issue. I updated the api response anyway to fix this

File: src/views/Connectors/Sources/SourcesForm/SourceConfigForm/MWTitleField.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { FormContextType, RJSFSchema, StrictRJSFSchema, TitleFieldProps } from '@rjsf/utils';
+import { Box, Divider, Heading } from '@chakra-ui/react';
+
+export default function MWTitleField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
```
[the remark this answers is not in the document]
Comment (lallenfrancisl) [reply src/views/Connectors/Sources/SourcesForm/SourceConfigForm/MWTitleField.tsx:4]: Here any is actually valid since it can be anything. That generic type goes in as the type for formdata. Since this is a generic component used everywhere, I don't think its possible to type this thing here

File: src/chakra.config.ts [hunk 5]
```diff
@@ -222,6 +224,23 @@ const extenstion = {
     heading: "Manrope, sans-serif",
     body: "Manrope, sans-serif",
   },
+  fontSizes: {
```
Comment (lallenfrancisl) [inline src/chakra.config.ts:227]: Added these to reflect the design

File: src/views/Connectors/NoConnectors/NoConnectors.tsx [hunk 6]
```diff
@@ -44,7 +44,7 @@ const NoConnectors = ({ connectorType }: NoConnectorsProps): JSX.Element => {
           </VStack>
           <Button
             onClick={() =>
-              navigate('/setup/destinations/new', { replace: true })
+              navigate('/setup/sources/new', { replace: true })
```
Comment (lallenfrancisl) [inline src/views/Connectors/NoConnectors/NoConnectors.tsx:47]: This is was a typo that slipped in some other commit

File: src/chakra.config.ts [hunk 7]
```diff
@@ -222,6 +224,23 @@ const extenstion = {
...
+    b3: "1rem",
+    b4: "0.875rem",
+    b5: "0.75rem",
```
Comment (sumitd94) [inline src/chakra.config.ts:230]: did we verify these token names from designs?

File: src/views/Connectors/NoConnectors/NoConnectors.tsx (hunk 6 above)
Comment (sumitd94) [reply src/views/Connectors/NoConnectors/NoConnectors.tsx:47]: If there are no connections, why do we want to take them to a new source? 🤔 
I believe this was right only.

Comment (lallenfrancisl) [reply src/views/Connectors/NoConnectors/NoConnectors.tsx:47]: This is when we click add source button in the empty screen, then sources form have to be opened instead of the destination form

Comment (lallenfrancisl) [reply src/views/Connectors/NoConnectors/NoConnectors.tsx:47]: Oh missed that this is used in destinations too. You are right, adding support for tha

Comment (sumitd94) [reply src/views/Connectors/NoConnectors/NoConnectors.tsx:47]: Thanks

**Review by sumitd94 — APPROVED** [review]
